client/extensions/woocommerce/app/settings/payments/stripe/payment-method-stripe-key-based-dialog.js

```diff
@@ -0,0 +1,240 @@
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import APIKeysView from 'woocommerce/components/api-keys-view';
+import AuthCaptureToggle from 'woocommerce/components/auth-capture-toggle';
+import Dialog from 'components/dialog';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextInput from 'components/forms/form-text-input';
+import { hasStripeKeyPairForMode } from './payment-method-stripe-utils.js';
+import PaymentMethodEditFormToggle from '../payment-method-edit-form-toggle';
+import TestLiveToggle from 'woocommerce/components/test-live-toggle';
+
+class PaymentMethodStripeKeyBasedDialog extends Component {
+
+	static propTypes = {
+		method: PropTypes.shape( {
+			settings: PropTypes.shape( {
+				apple_pay: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				capture: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				secret_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				publishable_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				testmode: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				test_publishable_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				test_secret_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+			} ),
+		} ),
+		onCancel: PropTypes.func.isRequired,
+		onEditField: PropTypes.func.isRequired,
+		onDone: PropTypes.func.isRequired,
+		onUserRequestsConnectFlow: PropTypes.func,
+		site: PropTypes.shape( {
+			domain: PropTypes.string.isRequired,
+		} ),
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			hadKeysAtStart: hasStripeKeyPairForMode( props.method ),
+			highlightEmptyRequiredFields: false,
+		};
+	}
+
+	renderHeading = () => {
+		const { translate } = this.props;
+
+		// If we had no keys at mount
+		if ( ! this.state.hadKeysAtStart ) {
+			return (
+				<div className="stripe__method-edit-header">
+					{ translate( 'Take credit card payments with Stripe' ) }
+				</div>
+			);
+		}
+
+		return (
+			<div className="stripe__method-edit-header">
+				{ translate( 'Manage Stripe' ) }
+			</div>
+		);
+	}
+
+	onSelectLive = () => {
+		this.props.onEditField( { target: { name: 'testmode', value: 'no' } } );
+	}
+
+	onSelectTest = () => {
+		this.props.onEditField( { target: { name: 'testmode', value: 'yes' } } );
+	}
+
+	renderModePromptAndKeyFields = () => {
+		const { method, translate } = this.props;
+
+		let keys = [];
+
+		if ( method.settings.testmode.value === 'no' ) {
+			keys = [
+				{
+					id: 'secret_key',
+					label: translate( 'Live Secret Key' ),
+					placeholder: translate( 'Enter your secret key from your Stripe.com account' ),
```

Comment:
These placeholder strings are the same in test and live modes. Should they be? And if so, maybe we can put them in a `const` to not repeat them here.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/17197/commits/4b5f156ddc505a0e4a4f4271a7531f008651485e